stan/math/prim/scal/meta/or.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef STAN_MATH_PRIM_SCAL_META_OR_HPP
+#define STAN_MATH_PRIM_SCAL_META_OR_HPP
+
+#include <type_traits>
+
+namespace stan {
+namespace math {
+
+/**
+ * A helper metaprogram to peform disjunction
+ * on type conditionals for zero or more provided types.
+ * See \link prim/scal/meta/contains_vector.hpp \endlink
+ * for an example.
+ */
+template <typename... Conds>
+struct or_ : std::false_type {};
+
+template <typename Cond, typename... Conds>
+struct or_<Cond, Conds...>
+    : std::conditional<Cond::value, std::true_type, or_<Conds...>>::type {};
+/**
+ * A helper metaprogram to peform disjunction
```

Comment:
Again, document what it does rather than describing it abstractly.  It defines a null class where the work's in the inherited type.

Reply:
`or_not_` was removed.